lithops/standalone/worker.py

```diff
@@ -41,7 +41,8 @@ def wait_job_completed(job_key):
     done = os.path.join(JOBS_DIR, job_key+'.done')
     while True:
         if os.path.isfile(done):
-            os.remove(done)
+# deleting the file prevents keeper detect job finish
+#            os.remove(done)
```

Comment:
Regarding this, I think there will be an issue here if the file is not deleted (on reuse mode) and a worker processes multiple tasks of the same job. Currently, keeper is designed to stop the VM once a single task of a given job is completed. The problem comes when a worker processes multiple tasks from the same job. Imagine tasks lasts for 60 seconds, and soft_dismantle is set to 10 seconds. If this file is not removed, the keeper will start soft dismantle after the first task, then the worker will get another task (of the same job) and put it into execution, however, the keeper will stop the VM during the execution of this 2nd task since this `JOB_KEY.done` file is already present on the VM.

Remove the file was a workaround to prevent this issue. But I think we should refactor the keeper to consider multiple tasks of the same job.

Reply:
Yep, I agree with your previous message with one minor addition: 

```
the worker spawned - keeper started and initiated to hard_dismantle timeout
the first task finished invocation on worker - keeper  updated to soft_dismatle timeout
the worker gets a new task - update keeper to hard_dismantle (because maybe soft_dismante is too low value and kills new tasks in the middle of the execution)
the second task finished - keeper updated again to soft dismantle timeout (timeout reset)
```

What do you think?

Reply:
Actually, that is exactly how it works now by design :)
When new task arrives, the keeper detects that there a running task and sets timeout to hard_dismantle timeout:

It happens here:
https://github.com/lithops-cloud/lithops/blob/395eef8b283512bd714d3633dcd94258e1df620c/lithops/standalone/keeper.py#L88

Reply:
There issue with master, that actually stays on soft timeout because it is not aware of tasks processing on workers, but I would prefer to deal with it in a scope of another PR.

Reply:
Yes, but when a worker receives tasks from different map jobs. If a worker receives tasks from the same job, then the keeper never changes to hard_dismantle. `JOB_KEY` [here](https://github.com/lithops-cloud/lithops/blob/395eef8b283512bd714d3633dcd94258e1df620c/lithops/standalone/keeper.py#L72) does not contain the task id (for eaxmple job_key is: `5f0bef-0-M000`), so it never detects there is a new task from the same job running on the worker, since a new task from the same job has the same JOB_KEY.

Reply:
You are right. Fixed. 
- Returned the job.done file delete in the wait_job_completed
- Update directly keeper job state to `done` in the wait_job_completed
- Removed job.done file check inside the keeper